creator-node/src/services/stateMachineManager/stateReconciliation/recoverOrphanedData.jobProcessor.ts

```diff
@@ -0,0 +1,57 @@
+import type {
+  DecoratedJobParams,
+  DecoratedJobReturnValue,
+  JobsToEnqueue
+} from '../types'
+import type {
+  RecoverOrphanedDataJobParams,
+  RecoverOrphanedDataJobReturnValue
+} from './types'
+
+const { QUEUE_NAMES } = require('../stateMachineConstants')
+
+/**
+ * Processes a job to find users who have data on this node but who do not have this node in their replica set.
+ * This means their data is "orphaned,"" so the job also merges this data back into the primary of the user's replica set and then wipes it from this node.
+ *
+ * @param {Object} param job data
+ * @param {Object} param.logger the logger that can be filtered by jobName and jobId
+ */
+module.exports = async function ({
+  logger
+}: DecoratedJobParams<RecoverOrphanedDataJobParams>): Promise<
+  DecoratedJobReturnValue<RecoverOrphanedDataJobReturnValue>
+> {
+  // TODO: Clear redis state
+
+  // TODO: Query this node's db to find all users who have data on it and save them in a redis set
```

Comment:
note - does node have sufficient memory to load whole set

Reply:
I'm going to loop through with pagination for the other set. for this one, I'm not sure. @dmanjunath any thoughts? maybe a LIMIT+OFFSET?

Reply:
i'd assume node.js doesn't but the VM does and paginate. plus i think any query we write from the discovery side should also be paginated so it can't be abused in either direction

Reply:
I think I'm clear on the discovery query with pagination, but I'm not sure I follow for the query against the CN's own postgres db -- are you saying we should do some kind of `models.CNodeUsers.findAll()` here but paginate that query as well?

Reply:
paginating DB query is important, but my point was that even if you paginate you're still aggregating all results into single process memory, vs processing one page at a time like monitor-state does

also reminder cursor pagination vs offset 🙂